### ISSUE TITLE
Dashboard assign groups issues

### ISSUE THREAD

**Issue by anupama2501** [rancher/dashboard#2568]
v2.5-fa087c81628de057406608f077d0952bb03cea56-head

1. From cluster explorer --> Users and authentication --> groups --> assign values
2. Assign a role with global permissions and create
3. Verify the group is listed 
4. Assign another group with global permissions selected for that group and create
5. Verify it is listed in the groups
4. Now from ember UI, global --> groups --> assign global roles --> select group to add with some global permissions enabled and click on create.
5. From the cluster explorer, if we refresh the page, the group assigned from ember UI and in the step 4 are not shown in the cluster explorer list of groups but the group created from step 2 is listed.

**cluster explorer**
<img width="200" alt="Screen Shot 2021-03-18 at 5 16 00 PM" src="https://user-images.githubusercontent.com/38144301/111713973-dbe36f80-880d-11eb-9b0f-d046e42a3605.png">

**ember UI:**
<img width="200" alt="Screen Shot 2021-03-18 at 5 16 06 PM" src="https://user-images.githubusercontent.com/38144301/111713979-e30a7d80-880d-11eb-8f11-c445183f0282.png">

6. Now if we assign another group from the cluster explorer and create, all the previously created groups even the one created from ember UI are listed in the groups page.

<img width="200" alt="Screen Shot 2021-03-18 at 4 27 03 PM" src="https://user-images.githubusercontent.com/38144301/111710452-f6661a80-8806-11eb-9a30-edc399f7c70c.png">

https://github.com/rancher/dashboard/issues/2530

**Comment by richard-cox** [comment]
@anupama2501 I haven't been able to reproduce this, does it occur every time given the above steps to reproduce?

**Comment by anupama2501** [comment]
@richard-cox yes. 
From step "7.From the cluster explorer, if we refresh the page, the group assigned from ember UI and in the step 4 are not shown in the cluster explorer list of groups but the group created from step 2 is listed."

If we refresh the whole page in dashboard and not "refresh group memberships" button, the issue is seen

**Comment by anupama2501** [comment]
@richard-cox Also observed, once the groups are created as admin, if we go to the cluster manager or logout and login as admin, in the explorer --> Users and auth --> groups, all the created groups are disappearing. 
Screenshot for reference:
<img width="400" alt="Screen Shot 2021-03-20 at 11 23 49 PM" src="https://user-images.githubusercontent.com/38144301/111896081-4ba35700-89d4-11eb-9b04-b66ccf46fcc9.png">
If we logout & login /go to cluster manager and then verify the cluster explorer --> Users and auth --> groups:
<img width="400" alt="Screen Shot 2021-03-20 at 11 18 41 PM" src="https://user-images.githubusercontent.com/38144301/111896157-bb194680-89d4-11eb-9da5-8e9225d04287.png">

**Comment by richard-cox** [comment]
Regarding the first scenario
- Are the cluster explorer/dashboard/new ui and the cluster manager/ember ui/old/ui...
   - in the same tab in the same window?
   - in different tabs in the same window?
   - in different browsers?
- At any point is the user signed out, or do the cluster explorer and cluster manager instances have a different user signed in to either?
- Is the user/s signed in using an auth provider and not a local user?

Regarding the second scenario
- This is to be expected. The same session is shared between the cluster explorer and cluster dashboard. So signing in to one as a local admin will result in the other making request as a local admin. Groups are only visible when logging in via an auth provider, so the groups list will be depopulated.

**Comment by anupama2501** [comment]
@richard-cox 
``` 
Are the cluster explorer/dashboard/new ui and the cluster manager/ember ui/old/ui...
in the same tab in the same window?
in different tabs in the same window?
in different browsers?
```
The cluster explorer/dashboard is opened in the same tab. No other window/tab/browser is open for the same session
The users are not signed out, from the same user, assigned more than 1 global role from cluster explorer --> Users and auth providers --> groups
In the same tab, clicked on the cluster manager, and created a group from the cluster manager global --> security --> groups
Now if we go back to cluster explorer and verify cluster explorer --> Users and auth providers --> groups [some groups created from the cluster explorer are seen missing]

```
Regarding the second scenario

This is to be expected. The same session is shared between the cluster explorer and cluster dashboard. 
So signing in to one as a local admin will result in the other making request as a local admin. 
Groups are only visible when logging in via an auth provider, so the groups list will be depopulated.
``` 

This is if the _**same user**_ logged out and logged in, the groups are disappearing. "Admin" is the auth provider Administrator not local admin. Although this is not always seen

**Comment by richard-cox** [comment]
I've test the two scenarios on the instance provided using the following steps and it all seems to work ok

Scenario 1
1. Log in as testuser1
1. Nav to `Global` / `local` cluster, click on `Cluster Explorer`
1. Nav to `Users & Authentication` / `Groups`,
1. click on `Assign Global Roles`
1. Select the `testgroup1` group, check `Administrator`, `Restricted Administrator` and `User-Base` (`Standard User` is already selected), click on `Save` (groups table shows `testgroup1` has `Admin`, `Restricted Admin`, `User` and `User Base` roles)
1. Click on `Assign Global Roles`
1. Select the `test(group&)` group, check `Administrator`, `Restricted Administrator` and `User-Base` (`Standard User` is already selected), click on `Save` (groups table shows `test(group&)` has `Admin`, `Restricted Admin`, `User` and `User Base` roles)
1. Select `Cluster Manager`
1. Select `Security` / `Groups`
1. Click on `Assign Global Roles`
1. Select the `_Group_Underscore` group, check `User-Base` and then `Create` (groups table shows `_Group_Underscore` has `User` and `User Base` roles)
1. Confirm all rows are still as expected
1. Repeat steps 2-3
1. Groups table shows correct values

Scenario 2
1. Log in as testuser1
1. Nav to `Global` / `local` cluster, click on `Cluster Explorer`
1. Nav to `Users & Authentication` / `Groups`,
1. click on `Assign Global Roles`
1. Select the `testgroup1` group, check `Configure Authentication`, click on `Save` (groups table shows `testgroup1` has `Manage Authentication` and `User` roles)
1. Click on `Assign Global Roles`
1. Select the `test(group&)` group, check `Use Catalogs`, click on `Save` (groups table shows `test(group&)` has `Use Catalog Templates` and `User` roles)
1. Click on User avatar and then logout
1. Repeat steps 1-3
1. Groups table shows correct values

Let's run through this together today

**Comment by richard-cox** [comment]
I believe the root issue was due to the call which populates groups (`/v3/principals`) missing some principals that had bindings. This has been fixed by #2601 & #2607

**Comment by anupama2501** [comment]
Verified on rancher commit ID: v2.5-bdec2f2a84e6dc5bca61b727ca5306174c816f33-head
Steps:

- rancher local cluster explorer --> Users and Authentication --> Auth providers --> Enable Auth [testuser1 as admin]
- rancher local cluster explorer --> Users and Authentication --> groups --> Assign global roles --> Add a member with permissions restricted admin, standard user --> Roles Create RKE Template Revisions, Create RKE Templates
- Add another member with permissions User Base, Roles --> Manage Node Drivers, Manage Settings, Authentication
- click on cluster manager in the top right corner of cluster explorer --> global --> security --> groups --> assign global roles and add a member as standard user permissions and roles authentication
- From cluster explorer verified all the roles and permissions for the group added in cluster manager is seen
- Logout and login as testuser1 and from cluster explorer groups verify all the groups created are shown
- Login as a standard user of one of the groups created in step 3and create a cluster and verify the cluster is created
